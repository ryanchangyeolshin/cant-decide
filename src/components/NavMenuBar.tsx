import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { useAuth0 } from "@auth0/auth0-react";
import { AppBar, Toolbar, Button, Icon, IconButton, Typography } from '@material-ui/core';
import { Menu as MenuIcon, Person as PersonIcon } from "@material-ui/icons";
import Typed from 'react-typed';
import NavMenuBarType from '../types/NavMenuBarType';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    toolBar: {
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
      background: "#154A40",
      color: "white"
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    loggedIn: {
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between"
    },
    username: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "flex-end",
      fontSize: "0.8rem",
      fontWeight: 500,
    },
    personIcon: {
      marginRight: "10px"
    },
    typed: {
      fontWeight: 900,
      fontSize: "1.5rem"
    }
  }),
);

const NavMenuBar = ({ toggleDrawer }: NavMenuBarType) => {
  const classes = useStyles();
  const { loginWithRedirect, user, isAuthenticated } = useAuth0();

  return (
    <AppBar position="static">
      <Toolbar className={classes.toolBar}>
        <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu" onClick={toggleDrawer(true)}>
          <MenuIcon />
        </IconButton>
        <Typed
          className={classes.typed}
          strings={[
          "Let's eat McDonalds",
          "Let's eat Taco Bell",
          "Can't Decide..."
        ]}
          typeSpeed={40}
          backSpeed={50}
          smartBackspace
        />
        {
          isAuthenticated ? (
            <div className={classes.loggedIn}>
              <Icon className={classes.personIcon}><PersonIcon /></Icon>
              <Typography variant="h6" className={classes.username}>
                {user.name.substring(0, user.name.lastIndexOf("@"))}
              </Typography>
            </div>
          ) : (
            <Button className="login-button" color="inherit" onClick={() => loginWithRedirect()}>
              Login
            </Button>
          )
        }
      </Toolbar>
    </AppBar>
  );
}

export default NavMenuBar;