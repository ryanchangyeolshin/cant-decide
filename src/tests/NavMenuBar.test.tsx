import React from 'react';
import { render } from '@testing-library/react';
import NavMenuBar from '../components/NavMenuBar';

describe("This suit is to test the NavMenuBar component", () => {
  test('Snapshot of NavMenuBar', () => {
    const { asFragment } = render(<NavMenuBar />);
    expect(asFragment()).toMatchSnapshot()
  });
});
