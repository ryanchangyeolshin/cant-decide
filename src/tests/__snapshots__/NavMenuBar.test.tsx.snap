// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the NavMenuBar component Snapshot of NavMenuBar 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-2 MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-3 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
